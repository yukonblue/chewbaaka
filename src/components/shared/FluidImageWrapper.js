/**
 * FluidImageWrapper.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 30, 2020
 * Updated  : Aug 23, 2021
 */

import React from 'react'

import './ContentPageSharedStyles.css'

export default function FluidImageWrapper({ src, alt, centered, width, height }) {
  let classNames = ["FullWidth", "DisplayBlock", "RetainAspectRatio"];

  if ( centered ) {
    classNames.push("Centered");
  }

  const aspectRatio = {
    width: parseInt(width, 10),
    height: parseInt(height, 10)
  }

  return (
    <img
      className={classNames.join(" ")}
      src={src}
      alt={alt}
      // specify aspect ratio
      width={aspectRatio.width}
      height={aspectRatio.height}
    />
  );
}
