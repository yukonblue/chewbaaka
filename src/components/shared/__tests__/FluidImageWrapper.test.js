/**
 * FluidImageWrapper.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 17, 2020
 * Updated  : Aug 23, 2021
 */

import React from 'react'

import { render } from '@testing-library/react'

import renderer from 'react-test-renderer'

import FluidImageWrapper from '../FluidImageWrapper'

const href = "https://google.com/image.jpg";
const src = "Some image";
const width = 520;
const height = 320;

test('renders FluidImageWrapper component', () => {
  render(
    <FluidImageWrapper
      src={src}
      href={href}
      width={width}
      height={height}
      centered
    />
  );
});

test('FluidImageWrapper component snapshot', () => {
  const tree = renderer
    .create(
      <FluidImageWrapper
        src={src}
        href={href}
        width={width}
        height={height}
        centered
      />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
