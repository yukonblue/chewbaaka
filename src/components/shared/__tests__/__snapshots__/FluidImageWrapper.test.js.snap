// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FluidImageWrapper component snapshot 1`] = `
<img
  className="FullWidth DisplayBlock RetainAspectRatio Centered"
  height={320}
  src="Some image"
  width={520}
/>
`;
